### ISSUE TITLE
API errors related to 'dev' CKAN instance

### ISSUE THREAD

**Issue by khaeru** [transport-data/tdc-data-portal#209]
Based on the information provided during the build-up phase for the portal, I merged transport-data/tools#35 to add CKAN Action API tools and tests to the TDCI Python tools package. This included adding the tests to daily scheduled GitHub Actions workflows.

[Here](https://github.com/transport-data/tools/actions/runs/12309885886/job/34357603967#step:7:4028) is an example from 2024-12-13 of these tests passing.

For the past couple of day some of these tests have been failing. I've seen errors such as [these](https://github.com/transport-data/tools/actions/runs/12346235268/job/34451519863#step:7:4964). To summarize, they seem to include:
- 404 Not Found responses for the /organization_show API endpoint.
- 403 Not Authorised for current_package_list_with_resources, organization_show, and package_search —this appears to be coming from Solr *via* CKAN; see the log.
- All for the 'dev' instance at https://ckan.tdc.dev.datopian.com

So some questions:
- Are these transient or intended changes?
  - If transient, might they reoccur? In this case I would use something like [pytest-rerunfailures](https://pytest-rerunfailures.readthedocs.io) to mark them as 'flaky', so that recurrences will not fail the test suite.
  - If intended, how could I adapt the existing tests for these endpoints? I.e. what would be queries that we should expect to always work?
- Do/will they also affect the production instance at ckan.transport-data.org (ckan.tdc.prod.datopian.com)?

Thanks in advance!

**Comment by luccasmmg** [comment]
@khaeru might i ask how did you grab the api key to run these tests?

**Comment by khaeru** [comment]
Currently these don't use any API token at all. From the PR I linked above: https://github.com/transport-data/tools/blob/a716994200b3fc1097a46cb2feca1b7ed474b822/transport_data/tests/util/test_ckan.py#L83-L156 —these marks reflect all the queries that appeared to work (and continued to work until a few days ago) without a token. (I didn't see any documentation that distinguished API endpoints that strictly require a token from those that do not.) All others are marked XFAIL, e.g. with `_NotAuthorized`.

Separately, I've generated my own API token for the production instance using the front-end. With help from @nicolas-becker, I was able to access to the dev instance front-end and also generate an API token there. I'll be adding code to use these (or any user's own tokens) in transport-data/tools#38. But I don't think this affects the tests and symptoms I'm referring to here.
